### DISCUSSION TITLE
Switching to Fastify

### DISCUSSION THREAD

**Discussion by NightScript370** [coder/code-server#2931] [Ideas] [answered]
Currently, code-server uses Express.JS, which is quite slow and bloated. This is bad because users on lower-end WiFi would have to load code-server really slowly. For example, I am stuck to using a mobile hotspot in school, due to the school WiFi using a whitelist system

Fastify is much quicker than Express by a large amount and the benchmark shows it: https://www.fastify.io/benchmarks/

I was wondering if anyone was interested in switching, or if I would be wasting my time if I were to make a Pull Request attempting this myself.

**Comment by jsjoeio** [comment, marked as answer]
This is the first time I've heard of Fastify but it looks promising. And better performance would be nice. 

@code-asher would love to get your thoughts on this?

**Reply by oxy** [reply]
Sounds interesting, but requests/sec is most likely/almost definitely _not_ the speedbump they're hitting - they mentioned low-end WiFi/mobile hotspots, which indicates that there's more of a bandwidth/latency issue.

I don't think it would help out in a meaningful enough way in that scenario, or for our usecase in general - `code-server` is a mostly single-user application so reqs/second isn't something we need to optimize for, imo.

**Reply by nooobcoder** [reply]
Upgrading to a newer JS framework would'nt be tiresome? @jsjoeio 
I know frameworks can be nice, but adapting them is difficult. See, Angular for example, it grew like fire and now, nobody touches it.

**Reply by jsjoeio** [reply]
Yeah, agreed. I think @oxy made solid points. Probably wouldn't make sense for us to change.
